src/internet/subsonic/subsonicservice.cpp

```diff
@@ -411,13 +416,19 @@ void SubsonicLibraryScanner::OnGetAlbumListFinished(QNetworkReply* reply,
   }
 
   int albums_added = 0;
-  reader.readNextStartElement();
-  Q_ASSERT(reader.name() == "albumList2");
-  while (reader.readNextStartElement()) {
-    Q_ASSERT(reader.name() == "album");
-    album_queue_ << reader.attributes().value("id").toString();
-    albums_added++;
-    reader.skipCurrentElement();
+  if(!skip_read_albums) {
+    reader.readNextStartElement();
+    if(reader.name() != "albumList2") {
+      qLog(Warning) << "Subsonic parsing error: albumList2 tag expected. Aborting scan.";
+      return;
+    }
+
+    while (reader.readNextStartElement()) {
+      Q_ASSERT(reader.name() == "album");
```

Comment:
You shouldn't assert on invalid input, just recognise that it's invalid and give up parsing.